internal/db/psqldb.go

```diff
@@ -0,0 +1,244 @@
+			' || quote_literal(interval_start_date) || '::timestamp with time zone  as interval_start,
+			' || quote_literal(interval_end_date) || '::timestamp with time zone  as interval_end,
+			namespace,
+			round(SUM(pod_avg_usage_cpu_core), 2),
+			round(SUM(pod_request_cpu_core), 2),
+			round(SUM(pod_limit_cpu_core), 2),
+			round(SUM(pod_avg_usage_memory) / 1024 / 1024, 0),
+			round(SUM(pod_request_memory) / 1024 / 1024, 0 ),
+			round(SUM(pod_limit_memory) / 1024 / 1024, 0 ),
+			round(SUM(volume_storage_request) / 1024 / 1024 / 1024, 2 ),
+			round(SUM(persistent_volume_claim_usage) / 1024 / 1024 / 1024, 2 ),
+			round(SUM(persistent_volume_claim_capacity) / 1024 / 1024 / 1024, 2 )
+			FROM
+			(SELECT
+				namespace,
+				(SUM(pod_usage_cpu_core_seconds) / ' || quote_literal(total_seconds) || ' * 1000 )::numeric      as pod_avg_usage_cpu_core,
+				(MAX(pod_request_cpu_core_seconds) / 3600 * 1000 )::numeric   as pod_request_cpu_core,
+				(MAX(pod_limit_cpu_core_seconds) / 3600 * 1000 )::numeric      as pod_limit_cpu_core,
+				(SUM(pod_usage_memory_byte_seconds)   / ' || quote_literal(total_seconds) || ')::numeric as pod_avg_usage_memory,
+				(MAX(pod_request_memory_byte_seconds) / 3600 )::numeric as pod_request_memory,
+				(MAX(pod_limit_memory_byte_seconds)   / 3600 )::numeric as pod_limit_memory,
+				(MAX(volume_request_storage_byte_seconds)   / 3600 )::numeric as volume_storage_request,
+				(MAX(persistentvolumeclaim_capacity_byte_seconds)   / 3600 )::numeric as persistent_volume_claim_capacity,
+				(SUM(persistentvolumeclaim_usage_byte_seconds)  / ' || quote_literal(total_seconds) || ')::numeric as persistent_volume_claim_usage
+			FROM (SELECT namespace, pod, interval_start, pod_usage_cpu_core_seconds, pod_request_cpu_core_seconds, pod_limit_cpu_core_seconds,
+					pod_usage_memory_byte_seconds, pod_request_memory_byte_seconds, pod_limit_memory_byte_seconds,
+					0 AS volume_request_storage_byte_seconds, 0 AS persistentvolumeclaim_capacity_byte_seconds, 0 AS persistentvolumeclaim_usage_byte_seconds
+					FROM logs_2
+					UNION ALL
+					SELECT namespace, pod, interval_start, 0.0 AS pod_usage_cpu_core_seconds, 0.0 AS pod_request_cpu_core_seconds,
+					0.0 AS pod_limit_cpu_core_seconds, 0.0 AS pod_usage_memory_byte_seconds, 0.0 AS pod_request_memory_byte_seconds,
+					0.0 AS pod_limit_memory_byte_seconds, volume_request_storage_byte_seconds, persistentvolumeclaim_capacity_byte_seconds,
+					persistentvolumeclaim_usage_byte_seconds FROM logs_3) AS temp
+			WHERE temp.interval_start >= ' || quote_literal(interval_start_date) || ' and temp.interval_start < ' || quote_literal(interval_end_date) || '
+			GROUP BY (namespace, pod))
+			AS t
+			GROUP BY namespace returning *';
+		end; $$ LANGUAGE plpgsql`
+
+	err := createTablesHelper(ctx, db, logs_0)
+	if err != nil {
+		return false, err
+	}
+
+	err = createTablesHelper(ctx, db, logs_1)
+	if err != nil {
+		return false, err
+	}
+
+	err = createTablesHelper(ctx, db, logs_2)
```

Comment:
It is suggested to make a single function call for tables and one more function call for generate function. Instead of making multiples. Rajath is pushing that change.

Reply:
Pushed the changes.